src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/EditApprenticeship.cshtml

```diff
@@ -168,51 +169,50 @@
             }
             else
             {
-                @if (false) //TODO re-add this post MVS AuthorizationService.IsAuthorized(SFA.DAS.ProviderCommitments.Features.ProviderFeature.DeliveryModel))
+                @if (AuthorizationService.IsAuthorized(SFA.DAS.ProviderCommitments.Features.ProviderFeature.DeliveryModel))
                 {
-                    <div class="govuk-form-group @Html.AddClassIfPropertyInError(x => x.DeliveryModel, "govuk-form-group--error")">
-                        <fieldset class="govuk-fieldset">
-                            <legend class="govuk-fieldset__legend govuk-fieldset__legend--s">
-                                Delivery model
-                            </legend>
-                            @Html.ValidationMessageFor(m => m.DeliveryModel, null, new { @class = "govuk-error-message", id = "error-message-" + Html.IdFor(m => m.DeliveryModel) })
-                            <div class="govuk-radios">
-                                <div class="govuk-radios__item">
-                                    <input type="radio" asp-for="DeliveryModel" class="govuk-radios__input" value="@DeliveryModel.Regular" id="DeliveryModelNormal" aria-describedby="DeliveryModelNormal-hint">
-                                    <label for="DeliveryModelNormal" class="govuk-label govuk-radios__label">
-                                        Regular
-                                    </label>
-                                    <div id="DeliveryModelNormal-hint" class="govuk-hint govuk-radios__hint">
-                                        The apprentice will have a single employer
-                                    </div>
-                                </div>
-                                <div class="govuk-radios__item">
-                                    <input type="radio" asp-for="DeliveryModel" class="govuk-radios__input" value="@DeliveryModel.PortableFlexiJob" id="DeliveryModelFlexible" aria-describedby="DeliveryModelFlexible-hint">
-                                    <label for="DeliveryModelFlexible" class="govuk-label govuk-radios__label">
-                                        Portable flexi-job
-                                    </label>
-                                    <div id="DeliveryModelFlexible-hint" class="govuk-hint govuk-radios__hint">
-                                        The apprentice will move between multiple employers
-                                    </div>
-                                </div>
-                            </div>
-                        </fieldset>
+                    if (!string.IsNullOrWhiteSpace(Model.CourseCode))
+                    {
+                        Model.CourseName = (await CommitmentsApiClient.GetTrainingProgramme(Model.CourseCode)).TrainingProgramme.Name;
```

Comment:
Do you think it would be better to move API calls and related logic into the controller / services. Ideally the view code is very simple with minimal logic and no processing.

I don't think we should do this here.

Reply:
Ideally yes, but it becomes a bit messy to do this. 

We have to add new mappers to simply populate the course name. This is because we are just changing the course code when re-selecting the course. 

We are already already doing this on the Draft Apprenticeship pages, so the pattern already exists.

Reply:
Even moving this into the controller would be a big improvement.  We don't need to introduce mappers / other layers if they're not already there,

Reply:
But we are already doing this in the in the Draft Apprenticeship page. And this solution is the least intrusive approach.